src/main/java/frc/robot/OI.java

```diff
@@ -0,0 +1,242 @@
+    if (povButton.getType() != Control.ControlType.POVBUTTON) {
+      DriverStation.reportWarning(povButton.getAction() + " is not a POV button", true);
+    }
+    return new POVButton(povButton.getController(), povButton.getId());
+  }
+
+  public static Trigger getTrigger(Control trigger) {
+    // "Trigger" referring to the type of button, not the WPI class
+    if (trigger.getType() != Control.ControlType.TRIGGER) {
+      DriverStation.reportWarning(trigger.getAction() + " is not a trigger", true);
+      return new Trigger(() -> false);
+    }
+    return new Trigger(
+        () -> trigger.getController().getRawAxis(trigger.getId()) > trigger.getThreshold());
+  }
+
+  private static class Control {
+    private enum ControlType {
+      AXIS,
+      BUTTON,
+      POVBUTTON,
+      TRIGGER
+    }
+
+    private int id;
+    private String action;
+    private String name; // Refers to button name
+    private XboxController controller;
+    private ControlCurve curve;
+    private double threshold; // Percentage where axis is triggered as a button
+    private ControlType type;
+
+    private Control(
+        int id, String action, String name, XboxController controller, ControlType type) {
+      this.id = id;
+      this.action = action;
+      this.name = name;
+      this.controller = controller;
+      this.type = type;
+      putControl();
+    }
+
+    Control(
+        XboxController.Axis axis, String action, XboxController controller, ControlCurve curve) {
+      this(axis.value, action, axis.name(), controller, ControlType.AXIS);
+      this.curve = curve;
+    }
+
+    Control(XboxController.Axis axis, String action, XboxController controller) {
+      this(axis, action, controller, new ControlCurve(1, 0, 0, 0));
+    }
+
+    Control(XboxController.Button button, String action, XboxController controller) {
+      this(button.value, action, button.name(), controller, ControlType.BUTTON);
+    }
+
+    Control(double povAngle, String action, XboxController controller) {
+      this((int) povAngle, action, "POV " + povAngle, controller, ControlType.POVBUTTON);
+    }
+
+    Control(XboxController.Axis axis, String action, XboxController controller, double threshold) {
+      this(axis.value, action, axis.name(), controller, ControlType.TRIGGER);
+      this.threshold = threshold;
+    }
+
+    private int getId() {
+      return id;
+    }
+
+    private String getAction() {
+      return action;
+    }
+
+    private String getName() {
+      return name;
+    }
+
+    private XboxController getController() {
+      return controller;
+    }
+
+    private ControlCurve getCurve() {
+      return curve;
+    }
+
+    private double getThreshold() {
+      return threshold;
+    }
+
+    private ControlType getType() {
+      return type;
+    }
+
+    private void putControl() {
+      if (controller.getPort() == driverJoystickPort) {
+        driverControlsLayout.add(
+            "Driver" + String.valueOf(getId()),
```

Comment:
This doesn't work because you Control id's are not unique. e.g. there is an axis of id 1 (Left Y axis) and a button of id 1 (A button).